### PR TITLE
doc: Fixed the documentation in wifi-subsys

### PR DESCRIPTION
**Pull request by xkevin190**
### Contribution description
the wifi subsys documentation was modified to have a better order and make it much more understandable for users.

### Testing procedure

cd /m4a-firmware 

`make docs`

### CODE REVIEW

File: wifi-subsys/components/protocols/doc.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+# Overview
+
+In this module are all the protocols used by wifi-subys, the protocols used are:
+
+1) icmp: this is used to check if esp32 is connected.
+
+2) http_client: the http client will be used to send the data that arrives from the peripherals to the server.
+
+3) sntp: this protocol is used to synchronize the time on the sp32
```
Comment (eduazocar) [inline wifi-subsys/components/protocols/doc.txt:14]: esp32

Comment (eduazocar) [inline wifi-subsys/components/protocols/doc.txt:14]: change the sp32 to esp32 or m4a-24g board reference

**Review by eduazocar — CHANGES_REQUESTED** [review]
Everything Looks good for me, Fix this little error and then do squash to approve this :+1:

**Comment by eduazocar** [comment]
Another thing, I see that appears some static-test failing, Use clang to fix it

File: wifi-subsys/components/protocols/doc.txt (hunk 1 above)
Comment (xkevin190) [reply wifi-subsys/components/protocols/doc.txt]: wifi-subsys only for esp32 bro.

**Review by eduazocar — APPROVED** [review]
Good Job :+1:

**Pull request merged**